src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -150,10 +182,119 @@ private async Task<string> GetFormattedDiff(RenderedCodeFile renderedCodeFile, R
+        }
+
+        private async Task<ReviewModel> GetBaseLineReview(string Language, string packageName, PullRequestModel pullRequestModel)
+        {
+            // Get  previously cloned review for this pull request or automatically generated master review for package
+            ReviewModel review;
+            if (pullRequestModel.ReviewId != null)
+            {
+                review = await _reviewsRepository.GetReviewAsync(pullRequestModel.ReviewId);
+            }
+            else
+            {
+                var autoReview = await _reviewsRepository.GetMasterReviewForPackageAsync(Language, packageName);
+                review = CloneReview(autoReview);
+                pullRequestModel.ReviewId = review.ReviewId;
+            }
+            return review;
+        }
+
+        private ReviewModel CloneReview(ReviewModel review)
+        {
+            var baseRevision = review.Revisions.Last();
+            var reviewCopy = new ReviewModel()
+            {
+                Author = review.Author,
+                CreationDate = baseRevision.CreationDate,
+                Name = review.Name,
+                IsAutomatic = false,
+                IsClosed = false,
+                RunAnalysis = review.RunAnalysis,
+                ReviewId = IdHelper.GenerateId()
+            };
+            reviewCopy.Revisions.Add(baseRevision);
+            return reviewCopy;
+        }
+
+        public async Task CleanupPullRequestData()
+        {
+            var telemetry = new RequestTelemetry { Name = "Cleaning up Reviews created for pull requests" };
+            var operation = _telemetryClient.StartOperation(telemetry);
+            try
+            {               
+                var pullRequests = await _pullRequestsRepository.GetPullRequestsAsync(true);
+                foreach(var prModel in pullRequests)
+                {
+                    try
+                    {
+                        if (await IsPullRequestEligibleForCleanup(prModel))
+                        {
+                            _telemetryClient.TrackEvent("Closing review created for pull request " + prModel.PullRequestNumber);
+                            await ClosePullRequestReview(prModel);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        _telemetryClient.TrackEvent("Failed to close review " + prModel.ReviewId);
+                        _telemetryClient.TrackException(ex);
+                    }
+                }                
+            }
+            catch(Exception ex)
+            {
+                _telemetryClient.TrackException(ex);
+            }
+            finally
+            {
+                _telemetryClient.StopOperation(operation);
+            }
+        }
+
+        private async Task<bool> IsPullRequestEligibleForCleanup(PullRequestModel prModel)
+        {
+            var repoInfo = prModel.RepoName.Split("/");
```

Comment:
May need to do some split validation here as well, or perhaps do the split as part of the creation of the PRModel instead.

Reply:
Added validation in controller to verify query params